nanofi/src/api/nanofi.cpp

```diff
@@ -371,9 +371,18 @@ int8_t remove_attribute(flow_file_record *ff, const char *key) {
 int get_content(const flow_file_record* ff, uint8_t* target, int size) {
   NULL_CHECK(0, ff, target);
   auto content_repo = static_cast<std::shared_ptr<minifi::core::ContentRepository>*>(ff->crp);
-  std::shared_ptr<minifi::ResourceClaim> claim = std::make_shared<minifi::ResourceClaim>(ff->contentLocation, *content_repo);
-  auto stream = (*content_repo)->read(claim);
-  return stream->read(target, size);
+  if(ff->crp && (*content_repo)) {
+    std::shared_ptr<minifi::ResourceClaim> claim = std::make_shared<minifi::ResourceClaim>(ff->contentLocation,
+                                                                                           *content_repo);
+    auto stream = (*content_repo)->read(claim);
+    return stream->read(target, size);
+  } else {
```

Comment:
The problem here is that ff->contentlocation is used for two different purpose:
-A simple filename to show where the content of the flowfile in the filesystem is
-An internal string that represent the location of the file within the content repo

I think this should be separated.
@phrocker do you have any longer term plan to resolve this in nanofi?

Reply:
Imagine if there were no c++ involved, and as a result, no content repo, would that change your concern?

Reply:
I wasn't aware of the nanofi related visions here.
In case we don't plan to support content repo in Nanofi, just have flow files store content in a simple file, that's fine. Although I would expect people to in IoT world to use nanofi to trasmit data without any disk i/o operations. 
As this struct is most probably going to change anyway in the near future, it's fine as is.